documentation/src/main/asciidoc/user_guide/streams.adoc

```diff
@@ -412,12 +412,20 @@ method on the `CacheStream`.  This value will default to the `chunkSize` configu
 Unfortunately this value is a tradeoff with memory usage vs performance vs at least once and your
 mileage may vary.
 
-*Using `iterator` with a replication cache*
+*Using `iterator` with cache replication*
+
+For replicated caches, {brandname} performs the `iterator` or `spliterator` terminal operations locally on the node, which optimizes performance as remote iterations are more resource intensive.
```

Comment:
We should probably mention that this is done with a distributed cache as well somehow. Basically if the local node owns all the required segments it won't go remote for these operations. So the case of DIST if the numNodes >= numOwners it will always be local. Also if a subset of segments are requested it may stay local as well.

Reply:
Bear with me, please. When you say "all the required segments" does this mean all the segments that will be included in a given stream?

"Also if a subset of segments are requested it may stay local as well." - Is this just for distributed caches? Where nodes are the primary owner for a subset of segments that constitute a stream, the iteration might be performed locally. Is there any more definite guidance we can provide for this case? The use of "may/might" here seems rather vague.

Reply:
Technically a user can supply a subset of segments via this method https://docs.jboss.org/infinispan/10.0/apidocs/org/infinispan/BaseCacheStream.html#filterKeySegments(org.infinispan.commons.util.IntSet)

In most cases they will not, in which case it will stay local if numNodes >= numOwners for DIST and will be local for REPL (barring the write behind shared exception)

I said may because if they supply a subset of segments, it will only stay local if all the segments are owned (primary or backup) by the local node with DIST.

It might be better to not go that much into detail here though, that is what I was struggling with.